openforcefield/tests/test_topology.py

```diff
@@ -618,6 +618,203 @@ def test_to_from_openmm(self):
+
+        from openforcefield.tests.test_forcefield import create_ethanol
+        from openforcefield.topology import Molecule, Topology
+
+        topology = Topology()
+        topology.add_molecule(create_ethanol())
+        mol = Molecule.from_pdb_and_smiles(
+            get_data_file_path("systems/test_systems/1_ethanol.pdb"), "CCO"
+        )
+        positions = mol.conformers[0]
+        count = 1
+        with NamedTemporaryFile(suffix=".pdb") as iofile:
+            topology.to_file(iofile.name, positions, file_format="pDb")
+            data = open(iofile.name).readlines()
+            for line in data:
+                if line.startswith("HETATM") and count == 1:
+                    count = count + 1
+                    coord = line.split()[-6]
+        assert coord == "10.172"
+
+    @requires_rdkit
+    def test_to_file_fileformat_invalid(self):
+        """
+        Checks for invalid file format
+        """
+        from openforcefield.tests.test_forcefield import create_ethanol
+        from openforcefield.topology import Molecule, Topology
+
+        topology = Topology()
+        topology.add_molecule(create_ethanol())
+        mol = Molecule.from_pdb_and_smiles(
+            get_data_file_path("systems/test_systems/1_ethanol.pdb"), "CCO"
+        )
+        positions = mol.conformers[0]
+        fname = "ethanol_file.pdb"
+        with pytest.raises(NotImplementedError):
+            topology.to_file(fname, positions, file_format="AbC")
+
+    def test_to_file_no_molecules(self):
+        """
+        Checks if Topology.to_write() writes a file with no topology and no coordinates
+        """
+        from tempfile import NamedTemporaryFile
+
+        from openforcefield.topology import Topology
+
+        topology = Topology()
+        lines = []
+        with NamedTemporaryFile(suffix=".pdb") as iofile:
+            topology.to_file(iofile.name, [])
+            data = open(iofile.name).readlines()
+            for line in data:
+                lines.append(line.split())
+        assert lines[1] == ["END"]
+
+    @requires_rdkit
+    def test_to_file_multi_molecule_different_order(self):
+        """
+        Checks for the following if Topology.to_write maintains the order of atoms
+         for the same molecule with different indexing
+        """
+        from tempfile import NamedTemporaryFile
+
+        from openforcefield.tests.test_forcefield import (
+            create_ethanol,
+            create_reversed_ethanol,
+        )
+        from openforcefield.topology import Molecule, Topology
+
+        topology = Topology()
+        topology.add_molecule(create_ethanol())
+        topology.add_molecule(create_reversed_ethanol())
+        mol = Molecule.from_pdb_and_smiles(
+            get_data_file_path("systems/test_systems/1_ethanol.pdb"), "CCO"
+        )
+        positions = mol.conformers[0]
+        positions = np.concatenate([positions, positions + 10.0 * unit.angstrom])
```

Comment:
```suggestion
        # Make up coordinates for the second ethanol by translating the first by 10 angstroms 
        # (note that this will still be a gibberish conformation, since the atom order in the second molecule is different)
        positions = np.concatenate([positions, positions + 10.0 * unit.angstrom])
```